src/object/cli_obj.c

```diff
@@ -3736,10 +3741,12 @@ obj_comp_cb(tse_task_t *task, void *data)
 			if (daos_handle_is_valid(obj_auxi->th) &&
 			    !(args->extra_flags & DIOF_CHECK_EXISTENCE) &&
 				 (task->dt_result == 0 ||
-				  task->dt_result == -DER_NONEXIST))
+				  task->dt_result == -DER_NONEXIST)) {
+				obj_addref(obj);
 				/* Cache transactional read if exist or not. */
-				dc_tx_attach(obj_auxi->th, DAOS_OBJ_RPC_FETCH,
-					     task);
+				dc_tx_attach(obj_auxi->th, obj,
```

Comment:
Yes, in theory we should check the return value, but because it is triggered in the RPC complete callback and the main operation has succeed successfully, we cannot fail the whole operation if dc_tx_attach() failed. Anyway, I think I can print out some error message when dc_tx_attach() failed in another patch.